cc/toolchain/cc_toolchain_config.bzl

```diff
@@ -0,0 +1,171 @@
+    compile_flags = [
+        "--target=" + target_system_name,
+        # Security
+        "-U_FORTIFY_SOURCE",  # https://github.com/google/sanitizers/issues/247
+        "-fstack-protector",
+        "-fno-omit-frame-pointer",
+        # Diagnostics
+        "-fcolor-diagnostics",
+        "-Wall",
+        "-Wthread-safety",
+        "-Wself-assign",
+    ]
+
+    # -fstandalone-debug disables options that optimize
+    # the size of the debug info.
+    # https://clang.llvm.org/docs/UsersManual.html#cmdoption-fstandalone-debug
+    dbg_compile_flags = ["-g", "-fstandalone-debug"]
+
+    opt_compile_flags = [
+        # No debug symbols.
+        "-g0",
+
+        # Aggressive optimizations, can increase binary size.
+        "-O3",
+
+        # Security hardening on by default.
+        "-D_FORTIFY_SOURCE=1",
+
+        # Removal of unused code and data at link time (can this increase
+        # binary size in some cases?).
+        "-ffunction-sections",
+        "-fdata-sections",
+    ]
+
+    cxx_flags = [
+        # The whole codebase should build with c++14
+        "-std=c++14",
+        # Use bundled libc++ for hermeticity
+        "-stdlib=libc++",
+    ]
+
+    link_flags = [
+        "--target=" + target_system_name,
+        "-lm",
+        "-no-canonical-prefixes",
+        # Below this line, assumes libc++ & lld
+        "-l:libc++.a",
+        "-l:libc++abi.a",
+        "-l:libunwind.a",
+        # Compiler runtime features.
+        "-rtlib=compiler-rt",
+        # To support libunwind
+        # It's ok to assume posix when using this toolchain
+        "-lpthread",
+        "-ldl",
+    ]
+
+    # linux/lld only!
+    link_flags.extend([
+        "-fuse-ld=lld",
+        "-Wl,--build-id=md5",
+        "-Wl,--hash-style=gnu",
+        "-Wl,-z,relro,-z,now",
+    ])
+
+    # Similar to link_flags, but placed later in the command line such that
+    # unused symbols are not stripped.
+    link_libs = []
+
+    # linux/lld only
+    opt_link_flags = ["-Wl,--gc-sections"]
+
+    # Unfiltered compiler flags; these are placed at the end of the command
+    # line, so take precendence over any user supplied flags through --copts or
+    # such.
+    unfiltered_compile_flags = [
+        # Do not resolve our symlinked resource prefixes to real paths.
+        "-no-canonical-prefixes",
+        # Reproducibility
+        "-Wno-builtin-macro-redefined",
+        "-D__DATE__=\"redacted\"",
+        "-D__TIMESTAMP__=\"redacted\"",
+        "-D__TIME__=\"redacted\"",
+        # Grailbio uses this but in not sure it's necessary, given clang seems to respect PWD.
+        # Will need to validate with some testing.
+        #"-fdebug-prefix-map={}=__bazel_toolchain_llvm_repo__/".format(toolchain_path_prefix),
+    ]
+
+    # Coverage flags:
+    coverage_compile_flags = ["-fprofile-instr-generate", "-fcoverage-mapping"]
+    coverage_link_flags = ["-fprofile-instr-generate"]
+
+    # true if using lld
+    supports_start_end_lib = True
```

Comment:
some of these might be able to be just passed as keyword args rather than defining a variable first but this is personal preference